libsel4muslcsys/src/sys_morecore.c

```diff
@@ -361,3 +361,9 @@ long sys_mmap2(va_list ap)
      * so this will not overflow */
     return sys_mmap_impl(addr, length, prot, flags, fd, offset * 4096);
 }
+
+long sys_munmap(va_list ap)
+{
+    ZF_LOGV("calling dummy version of sys_munmap()");
```

Comment:
```suggestion
    ZF_LOGW("sys_munmap() is unsupported");
```
Maybe better say that it's not supported. That message is more transparent than the other one with the obscure number. And as @kent-mcleod said, it might be worth having this as a warning, as the higher layers should be fixed to no try this. It might be a bit of a pain for porting, But another solution might be a config feature that allows disabling this warning then.